### PR TITLE
MPW Design Description

### LINKED ISSUE #32: The "designs" directory should have an index file
The [designs directory](https://github.com/The-OpenROAD-Project/OpenLane-MPW-CI/tree/main/designs) has quite a large number of designs but no information at all about them. It would be good to have a `README.md` file which described some information about each design.

Relevant information would be;
 * Name.
 * Short description.
 * Who made it.
 * Type of design.
 * Where / when it was taped out.
 * Link to the current status on CI.
 * Link to a dashboard which shows the metrics about the design over time.

### PR DESCRIPTION
**Pull request by vijayank88**
@mithro 
Design description for each MPW shuttle designs updated to resolve #32

### CODE REVIEW

**Comment by mithro** [comment]
As I mentioned previously, a person who views this repository shouldn't care about which MPW a design ran on.

**Comment by vvbandeira** [comment]
@mithro currently, we have ~90 designs for four shuttles, if we keep this avg for mpw-6 we will add 20+ more and so on for each new shuttle. If we keep the current organization we will soon have over 100 subfolders under designs (even more under scripts) and a README with well over 1k lines. Do you think the current organization is ok?

I was thinking about changing the organization, either by creating `mpw-X/design` and `mpw-X/scripts` folders on the root of the repo or a subfolder under the existing folders `design/mpw-X`, `scripts/mpw-X` and have the README for each folder/shuttle -- however this organization might be opposed to your statement:

> a person who views this repository shouldn't care about which MPW a design ran on.

**Comment by vijayank88** [comment]
> @mithro currently, we have ~90 designs for four shuttles, if we keep this avg for mpw-6 we will add 20+ more and so on for each new shuttle. If we keep the current organization we will soon have over 100 subfolders under designs (even more under scripts) and a README with well over 1k lines. Do you think the current organization is ok?
> 
> I was thinking about changing the organization, either by creating `mpw-X/design` and `mpw-X/scripts` folders on the root of the repo or a subfolder under the existing folders `design/mpw-X`, `scripts/mpw-X` and have the README for each folder/shuttle -- however this organization might be opposed to your statement:
> 
> > a person who views this repository shouldn't care about which MPW a design ran on.

@vvbandeira 
If you're planning to split `designs/mpw-x`, then current description can be re-used easily.

**Comment by mithro** [comment]
Which MPW run these designs happened to be submitted on should not be relevant, hence I'm against using that information in any way to organize them.

**Comment by vijayank88** [comment]
@mithro  

Updated design description in alphabetical order with commit [c37ff10](https://github.com/The-OpenROAD-Project/OpenLane-MPW-CI/pull/38/commits/c37ff10c8dde75313c086125c714a51980c4f489)

@vvbandeira  FYI

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: designs/MPW_Design_Description.md [hunk 1]
```diff
@@ -0,0 +1,857 @@
...
+         exact fabric description.
+    -   Executable CI Script: [eFPGA_v3_wrapper](../scripts/eFPGA_v3_wrapper)
+
+-   `FCNet_prj`
+    -   Project Name: pre-trained neural network for MNIST
+    -   Project Owner: mxiangyue
+    -   Project Number: [486](https://platform.efabless.com/projects/486)
+    -   Description: This project implements a pre-trained neural
+         network for hand-written digits from MNIST dataset.
+    -   Executable CI Script: [FCNet_prj](../scripts/FCNet_prj)
```
[the remark this answers is not in the document]
Comment (vijayank88) [reply designs/MPW_Design_Description.md:270]: its `../scripts/FCNet_neuron`

File: designs/MPW_Design_Description.md [hunk 2]
```diff
@@ -0,0 +1,857 @@
...
+
+-   `vsdbabysoc`
+    -   Project Name: VSDBabySoC
+    -   Project Owner: Mufutau Akuruyejo
+    -   Project Number: [519](https://platform.efabless.com/projects/519)
+    -   Description: VSDBabySoC is a small SoC including PLL, DAC, and a
+         RISCV-based processor named RVMYTH.
+    -   Executable CI Script: [vsdbabysoc](../scripts/vsdbaby_soc) |
+         [vsdbabysoc_wrapper](../scripts/vsdbabysoc_wrapper)
+
```
[the remark this answers is not in the document]
Comment (vijayank88) [reply designs/MPW_Design_Description.md:801]: its `vsdbabysoc`

File: designs/MPW_Design_Description.md [hunk 3]
```diff
@@ -0,0 +1,857 @@
...
+    -   Description: Two string version of Karplus-Strong Guitar
+    -   Executable CI Script: [ks-guitar-2s](../scripts/ks-guitar-2s)
+
+-   `logic_bist`
+    -   Project Name: Logic BIST
+    -   Project Owner: Dinesh Annaya
+    -   Project Number: [566](https://platform.efabless.com/projects/566)
+    -   Description: Logic BIST with Scan Chain to detect struck at
+         fault
+    -   Executable CI Script: [logic_bist](../scripts/logic_bist)
```
[the remark this answers is not in the document]
Comment (vijayank88) [reply designs/MPW_Design_Description.md:425]: Both are different project. `566` project need to be fixed locally. For time being I'll remove CI script path. Later I will make another PR this project.

File: designs/MPW_Design_Description.md [hunk 4]
```diff
@@ -0,0 +1,857 @@
...
+         by radiation etc. and assure reliable functionality.
+    -   Executable CI Script: [RAD_HARD_ALU](../scripts/RAD_HARD_ALU)
+
+-   `randsack`
+    -   Project Name: Randsack
+    -   Project Owner: Harrison Pham
+    -   Project Number: [451](https://platform.efabless.com/projects/451)
+    -   Description: Random number generators and PUFs. Also a few
+         simple peripherals to output the random values (PWM, etc).
+    -   Executable CI Script: [randsack](../scripts/randsack)
```
[the remark this answers is not in the document]
Comment (vijayank88) [reply designs/MPW_Design_Description.md:630]: scripts path updated

File: openlane-tag [hunk 5]
```diff
@@ -1 +1 @@
-2022.04.28_01.44.59
```
[the remark this answers is not in the document]
Comment (vijayank88) [reply openlane-tag:1]: Ok I will do that.

Is there a way to stop running CI for adding documents/README file.

File: scripts/FMAC_wrapper/setup-flow.sh [hunk 6]
```diff
@@ -1 +1,2 @@
 # copy config files to design dir
+sed -i 's/mprj/mac_core/g' ${design_dir}/openlane/user_project_wrapper/config.tcl
```
[the remark this answers is not in the document]
Comment (vijayank88) [reply scripts/FMAC_wrapper/setup-flow.sh:2]: Ok

File: openlane-tag (hunk 5 above)
[the remark this answers is not in the document]
Comment (vvbandeira) [reply openlane-tag:1]: There's no "skip ci" like in OpenLane, only killing the job after it starts. Please revert this change on your next commit.

File: scripts/FMAC_wrapper/setup-flow.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (vvbandeira) [reply scripts/FMAC_wrapper/setup-flow.sh:2]: Please revert this change on your next commit.

File: designs/MPW_Design_Description.md [hunk 7]
```diff
@@ -0,0 +1,851 @@
...
+    -   Project Name: LBIST-MBIST
+    -   Project Owner: Dinesh Annaya
+    -   Project Number: [771](https://platform.efabless.com/projects/771)
+    -   Description: Logic built-in self-test (or LBIST) is a form of
+         built-in self-test (BIST) in which hardware and/or software is
+         built into integrated circuits allowing them to test their own
+         operation.
+    -   Executable CI Script:
+         [LBIST_mbist_top1](../scripts/LBIST_mbist_top1) |
+         [LBIST_mbist_top2](../scripts/LBIST_mbist_top2)
```
Comment (vvbandeira) [inline designs/MPW_Design_Description.md:429]: Are these duplicated or is one missing from CI?

File: README.md [hunk 8]
```diff
@@ -30,6 +30,8 @@ Add the design repository as a submodule.
 git submodule add <PROJECT_URL> ./designs/<PROJECT_NAME>
 ```
 
+Also you can view current CI design description [here](./designs/README.md)
+
```
Comment (vvbandeira) [inline README.md:34]: Add a period at the end of the sentence && move to line 17.

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: designs/MPW_Design_Description.md (hunk 7 above)
Comment (vijayank88) [reply designs/MPW_Design_Description.md:429]: Its not duplicate. Need to update 566 designs into CI later.

Comment (vijayank88) [reply designs/MPW_Design_Description.md:429]: consider its re-submitted design with improvement of designs from previous MPW shuttle to current one

Comment (vijayank88) [reply designs/MPW_Design_Description.md:429]: Its re-subitted MPW shuttle designs. Project Name and number will take to MPW shuttle path. But currently we are using latest designs in the CI

**Comment by vijayank88** [comment]
@mithro 
Please review and we can close this PR.

**Comment by mithro** [comment]
All the CI tests are failing?

**Review by mithro — APPROVED** [review]
README changes look good now!

**Pull request merged**